backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ConnectionManager.java

```diff
@@ -165,6 +168,50 @@ public void acceptConnection(@NonNull String connectionId) {
         }
     }
 
+    @Inject
+    InvitationParser invitationParser;
+
+    public CheckInvitationResponse checkReceivedInvitation(@NonNull String invitationUrl) {
+        return invitationParser.checkInvitation(invitationUrl);
+    }
+
+    public void receiveInvitation(@NonNull String encodedInvitation) {
+        InvitationParser.Invitation invitation = invitationParser.parseInvitation(encodedInvitation);
+        if (invitation.isParsed()) {
+            if (invitation.getInvitationRequest() != null) {
+                try {
+                    ac.connectionsReceiveInvitation(invitation.getInvitationRequest(),
+                            ConnectionReceiveInvitationFilter.builder()
+                                    .alias(invitation.getInvitationRequest().getLabel())
+                                    .autoAccept(true)
+                                    .build())
+                            .ifPresent(record -> {
+                                partnerRepo.save(Partner
+                                        .builder()
+                                        .ariesSupport(Boolean.TRUE)
+                                        .alias(StringUtils.trimToNull(record.getTheirLabel()))
+                                        .connectionId(record.getConnectionId())
+                                        .invitationMsgId(record.getInvitationMsgId())
+                                        .did(didPrefix + UNKNOWN_DID)
+                                        .state(ConnectionState.INVITATION)
+                                        .incoming(Boolean.TRUE)
+                                        .tags(null)
```

Comment:
It would be cool if we could keep everything consistent. If we generate a partner invitation we also have the possibility to set an alias and tags.